### PR TITLE
[RFC] dsl for model agnositc

### PR DESCRIPTION
**Pull request by frederick0329**
Dsl that compiles our model-agnostic analysis component.

cc @kubeflow/caipe-dev 

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/295)

### CODE REVIEW

File: samples/model-agnositc/model-analysis.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+import datetime
+
+
+def dataflow_tf_model_analyze_op(image: 'GcrUri', csv_eval_path_with_predict: 'GscUri', schema: 'GcsUri[text/json]', output: 'analysis_output', step_name='analysis'):
+    return mlp.ContainerOp(
```
Comment (sshrdp) [inline samples/model-agnositc/model-analysis.py]: What does mlp refer to here?

Comment (qimingj) [reply samples/model-agnositc/model-analysis.py]: It should be "dsl". Same below.

File: samples/model-agnositc/model-analysis.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+@dsl.pipeline(
+  name='TFMA Model Agnostic Example',
+  description='Example pipeline that does model analysis based on a csv file with features, predicitons, labels.'
+)
+def model_analysis(
+    output: dsl.PipelineParam,
+    project: dsl.PipelineParam,
+
+    image: mlp.PipelineParam=mlp.PipelineParam(
```
Comment (qimingj) [inline samples/model-agnositc/model-analysis.py]: If the default value is empty, then you don't need to provide a default value?

Comment (qimingj) [inline samples/model-agnositc/model-analysis.py]: Container image as a pipeline parameter is not supported yet. Can you hard code the container image here?

Comment (frederick0329) [reply samples/model-agnositc/model-analysis.py]: @salehay

**Comment by vicaire** [comment]
Looks like we lost track of this one, is it still needed?

**Comment by frederick0329** [comment]
Thanks this is not needed for now.

**Pull request closed without merging**